diff --git a/qga/commands-posix.c b/qga/commands-posix.c
index 0dc219d..9d020d3 100644
--- a/qga/commands-posix.c
+++ b/qga/commands-posix.c
@@ -102,6 +102,8 @@ void qmp_guest_shutdown(bool has_mode, const char *mode, Error **errp)
         reopen_fd_to_null(1);
         reopen_fd_to_null(2);

+        execle("/run/current-system/sw/bin/shutdown", "shutdown", "-h", shutdown_flag, "+0",
+               "hypervisor initiated shutdown", (char*)NULL, environ);
         execle("/sbin/shutdown", "shutdown", "-h", shutdown_flag, "+0",
                "hypervisor initiated shutdown", (char*)NULL, environ);
         _exit(EXIT_FAILURE);
@@ -159,3 +161,3 @@ void qmp_guest_set_time(bool has_time, int64_t time_ns, Error **errp)
     struct timeval tv;
-    static const char hwclock_path[] = "/sbin/hwclock";
+    static const char hwclock_path[] = "/run/current-system/sw/bin/hwclock";
     static int hwclock_available = -1;
